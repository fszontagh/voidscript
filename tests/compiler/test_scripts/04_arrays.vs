// Test array operations
int[] $numbers = [1, 2, 3, 4, 5];
string[] $fruits = ["apple", "banana", "cherry"];

printnl("Numbers array:");
for (int $num : $numbers) {
    printnl("  ", $num);
}

printnl("Fruits array:");
for (string $fruit : $fruits) {
    printnl("  ", $fruit);
}

// Array indexing
printnl("First number: ", $numbers[0]);
printnl("Second fruit: ", $fruits[1]);

// Array modification
$numbers[0] = 10;
$fruits[2] = "grape";

printnl("Modified first number: ", $numbers[0]);
printnl("Modified third fruit: ", $fruits[2]);

// Array length (if supported)
printnl("Numbers array has ", count($numbers), " elements");
printnl("Fruits array has ", count($fruits), " elements");