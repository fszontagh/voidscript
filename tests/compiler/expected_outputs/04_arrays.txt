Numbers array:
  1
  2
  3
  4
  5
Fruits array:
  apple
  banana
  cherry
First number: 1
Second fruit: banana
Modified first number: 10
Modified third fruit: grape
Numbers array has 5 elements
Fruits array has 3 elements